CHANGELOG.md

```diff
@@ -33,6 +33,11 @@
 * Fixed an issue where street names in spoken instructions could be go unpronounced if tagged with a [pronunciation](https://wiki.openstreetmap.org/wiki/Key:name:pronunciation). ([#3765](https://github.com/mapbox/mapbox-navigation-ios/pull/3765))
 * Added support for [the workaround](https://github.com/mapbox/mapbox-directions-swift/issues/662) for requesting a route that avoids an arbitrary set of coordinates. ([#3765](https://github.com/mapbox/mapbox-navigation-ios/pull/3765))
 
+### Routing
+
+* Integrated MaboxNavigtionNative rerouting mechanism. Rerouting customization is done using providing user implementation of `RoutingProvider` protocol to `NavigationService.init(routeResponse:routeIndex:routeOptions:routingProvider:credentials:locationSource:eventsManagerType:simulating:routerType:)` or `Router.init(alongRouteAtIndex:in:options:routingProvider:dataSource:)`. Passing `nil` will result in default behavior. ([#3754](https://github.com/mapbox/mapbox-navigation-ios/pull/3754))
```

Comment:
MaboxNavigationNative is an implementation detail. Certainly a rearchitecture of this scope would merit some mention in the changelog as a heads-up, but can we express it purely in terms of the impact on the public API and any expected user-facing changes?